src/java/htsjdk/variant/vcf/VCFHeaderLineTranslator.java

```diff
@@ -83,13 +83,29 @@
         // are we inside a quotation? we don't special case ',' then
         boolean inQuote = false;
 
+        // if we are in a quote and we see a backslash followed by quote, treat it as an escaped quote
+        boolean escape = false;
+
         // a little switch machine to parse out the tags. Regex ended up being really complicated and ugly [yes, but this machine is getting ugly now... MAD]
         for (char c: valueLine.toCharArray()) {
-            if ( c == '\"' ) {
-                inQuote = ! inQuote;
+            if ( c == '\"') {
+                if (escape) {
+                    builder.append(c);
+                    escape = false;
+                } else {
+                    inQuote = !inQuote;
+                }
             } else if ( inQuote ) {
-                builder.append(c);
+                if (escape) {
+                    builder.append(c);
+                    escape = false;
+                } else if (c != '\\') {
+                    builder.append(c);
+                } else {
+                    escape = true;
+                }
             } else {
+                escape = false;
                 switch (c) {
                     case ('<') : if (index == 0) break; // if we see a open bracket at the beginning, ignore it
                     case ('>') : if (index == valueLine.length()-1) ret.put(key,builder.toString().trim()); break; // if we see a close bracket, and we're at the end, add an entry to our list
```

Comment:
The VCF spec only mentions escaping in the context of the Description tags for INFO lines. However, it seems like the rules should also apply to other quoted text fields in the header, and these are causing problems in the wild. For example, GATK's VariantFiltration will add header lines with unescaped quotes to VCF headers. Issues samtools/hts-specs#89 and samtools/hts-specs#125 both address this lack of clarity in the spec. Is it OK to add this escaping capability in the meantime?